packages/core/src/cli/build.ts

```diff
@@ -0,0 +1,26 @@
+import { type RsbuildInstance, createRsbuild } from '@rsbuild/core';
+import { composeRsbuildEnvironments, pruneEnvironments } from '../config';
+import type { RslibConfig } from '../types/config';
+import { getAbsolutePath } from '../utils/helper';
+import type { BuildOptions } from './commands';
+
+export async function build(
+  config: RslibConfig,
+  options: Pick<BuildOptions, 'root' | 'lib' | 'watch'> = {},
+): Promise<RsbuildInstance> {
+  const cwd = process.cwd();
+  options.root = options.root ? getAbsolutePath(cwd, options.root) : cwd;
```

Comment:
`options.root ` has is calculated in `packages/core/src/cli/init.ts#10` and passed to hereby. Do we need to re-calculate it again? (so does mf).

Reply:
This `export async function build(
  config: RslibConfig,
  options: Pick<BuildOptions, 'root' | 'lib' | 'watch'> = {},
): Promise<RsbuildInstance>` function is aim to be our javascript api, so the absolute root path should be generated since `packages/core/src/cli/init.ts#10` is the cli logic. This file is put into the `cli` folder which may be not very good. Since we are not design the javascript api well, we can keep it here.

Reply:
It's better not to modify parameter to avoid potential bug.

Reply:
Agree. Anything async is very much prone to bugs. And like @fi3ework said, modifying parameters is a no-go.

Function parameters should remain **immutable**.


_Remember, UI = fn(data)_